.github/workflows/docker-ci.yml

```diff
@@ -0,0 +1,628 @@
+          else
+            cp $HOME/runner/artifacts/${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}/image.tar ${GITHUB_WORKSPACE}
+            echo "Copied image.tar from $HOME/runner/artifacts/${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}/image.tar to ${GITHUB_WORKSPACE}"
+          fi
+
+      - name: Setup rclone
+        uses: ./.github/actions/setup-rclone
+        with:
+          access_key_id: ${{ secrets.ACACIA_ACCESS_KEY_ID }}
+          secret_access_key: ${{ secrets.ACACIA_SECRET_ACCESS_KEY }}
+          endpoint: https://projects.pawsey.org.au
+          bucket: ${{ env.BUCKET }}
+          destination_path: ${{ env.DESTINATION_PATH }}
+
+      - name: Upload image tar to S3 with rclone
+        run: |
+          set -e
+          rclone copy ${{ github.workspace }}/image.tar pawsey0001:"${{ env.BUCKET }}/${{ env.DESTINATION_PATH }}/"\
+              --transfers=16 \
+              --drive-chunk-size=128M \
+              --multi-thread-streams=4   
+
+      - name: Login to Docker Hub       
+        uses: docker/login-action@v3
+        with:
+          username: ${{ vars.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      
+      - name: Load Docker image from image.tar
+        run: |
+            docker load -i ${GITHUB_WORKSPACE}/image.tar
+
+      - name: Tag Docker image for Dockerhub and Quay.IO
+        run: |
+          docker tag ${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}:${{ needs.PREPARE-job.outputs.date }} ${{ vars.DOCKERHUB_USERNAME }}/${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}:${{ needs.PREPARE-job.outputs.date }}
+
+      - name: Push Docker image to Dockerhub after approval
+        run: |          
+          docker push ${{ vars.DOCKERHUB_USERNAME }}/${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}:${{ needs.PREPARE-job.outputs.date }}
+
+      - name: Remove Docker images
+        run: |
+          # Remove the main image
+          docker rmi ${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}:${{ needs.PREPARE-job.outputs.date }} || true
+        
+            # Remove tagged images
+          docker rmi ${{ vars.DOCKERHUB_USERNAME }}/${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}:${{ needs.PREPARE-job.outputs.date }} || true
+
+
+  CLEANUP-job:
+    needs: [APPROVE-PUSH-PUB-job,PUSH-PRIV-job, SCAN-AND-REPORT-job, BUILD-job, PREPARE-job]
+    if: always()
+    runs-on: ${{ needs.PREPARE-job.outputs.runner_label }}
+    steps:
+      - name: Clean-up
+        run: |
+          sudo rm -rf $HOME/runner/artifacts/${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}
+
+  DEPLOY-job:
```

Comment:
This is one where I need to review the logic more. Singularity is available as a module on Setonix and typically we will look to have modules so you couldn't just run `singularity pull`. The `$MYSCRATCH` environment variable is also tricky. 

Maybe we think about not having this logic. Thoughts?

Reply:
Yes. Once merging, I think my first priority is modifying this part to adapt to data-mover instance environment for Setonix.

Also, do you think it is reasonable to run `tmux` which holds `github action runner` in the background on the data-mover instance? @pelahi  Do I need to apply some approval for long-time running service on data-mover as part of Setonix?

Reply:
For Ella, the automatic deployment is useful as it can save time for transferring image from Acacia to Ella as a smooth workflow.